src/test/java/org/broadinstitute/hellbender/tools/walkers/filters/VariantFiltrationIntegrationTest.java

```diff
@@ -0,0 +1,246 @@
+
+
+    @Test
+    public void testMaskReversed() throws IOException {
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString("vcfexample2.vcf", " -maskName outsideGoodSites -filterNotInMask --mask BED:" + getToolTestDataDir() + "goodMask.bed"),
+                Arrays.asList(getToolTestDataDir() + "expected/" + "testVariantFiltration_testMaskReversed.vcf")
+        );
+
+        spec.executeTest("test file", this);
+    }
+
+    @Test
+    public void testIllegalFilterName() throws IOException {
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString("vcfexample2.vcf", " -filter 'DoC < 20 || FisherStrand > 20.0' -filterName 'foo < foo' "),
+                1,
+                UserException.class
+        );
+
+        spec.executeTest("test file", this);
+    }
+
+    @Test
+    public void testFilter1() throws IOException {
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString("vcfexample2.vcf", " -filter 'DoC < 20 || FisherStrand > 20.0' -filterName foo "),
+                Arrays.asList(getToolTestDataDir() + "expected/" + "testVariantFiltration_testFilter1.vcf")
+        );
+
+        spec.executeTest("test file", this);
+    }
+
+    @Test
+    public void testFilter2() throws IOException {
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString("vcfexample2.vcf", " -filter 'AlleleBalance < 70.0 && FisherStrand == 1.4' -filterName bar "),
+                Arrays.asList(getToolTestDataDir() + "expected/" + "testVariantFiltration_testFilter2.vcf")
+        );
+
+        spec.executeTest("test file", this);
+    }
+
+    @Test
+    public void testFilterWithSeparateNames() throws IOException {
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString("vcfexample2.vcf", " --filterName ABF -filter 'AlleleBalance < 0.7' --filterName FSF -filter 'FisherStrand == 1.4' "),
+                Arrays.asList(getToolTestDataDir() + "expected/" + "testVariantFiltration_testFilterWithSeparateNames.vcf")
+        );
+
+        spec.executeTest("test file", this);
+    }
+
+    @Test
+    public void testInvertFilter() throws IOException {
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString("vcfexample2.vcf", " --filterName ABF -filter 'AlleleBalance < 0.7' --filterName FSF -filter 'FisherStrand == 1.4' --invertFilterExpression "),
+                Arrays.asList(getToolTestDataDir() + "expected/" + "testVariantFiltration_testInvertFilter.vcf")
+        );
+
+        spec.executeTest("test file", this);
+    }
+
+    @Test
+    public void testInvertJexlFilter() throws IOException {
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString("vcfexample2.vcf", " --filterName ABF -filter 'AlleleBalance >= 0.7' --filterName FSF -filter 'FisherStrand != 1.4' "),
```

Comment:
Is this meant to test `--invertFilterExpression`? (because it's not using that argument in its command line!)